packages/fxa-auth-server/lib/payments/stripe.ts

```diff
@@ -2354,15 +2409,25 @@ export class StripeHelper {
     };
   }
 
-  async extractCustomerDefaultPaymentDetails(uid: string) {
+  async extractCustomerDefaultPaymentDetails(
```

Comment:
Hmm the split params depending on the caller for this helper is confusing. Since `setCustomerLocation` isn't used anywhere yet, perhaps it could take a `uid` instead of the `customer` as a param to avoid this?

I do like the pattern of giving back various pieces of info related to the payment method and having the caller destructure to get whatever it needs.

Reply:
I anticipate the upcoming code will have a customer object, either from `stripe.customers.list` or a customer after a successful subscription.

We could simply have two functions.  The one that takes a uid would first fetch the customer then call the one that takes a customer.  What do you think?

Reply:
Sure that works too. Yeah I haven't thought about exactly where it will be called but I agree with you we will probably already have the Stripe customer.